src/deadline_worker_agent/scheduler/scheduler.py

```diff
@@ -668,6 +669,98 @@ def _fail_all_actions(
+                # been violated.
+                message = (
+                    "FATAL: Queue does not have a jobRunAsUser. This should not be possible. "
+                    "Please report this to the service team."
+                )
+                raise ValueError(message)
+            elif not job_run_as_user.is_worker_agent_user:
+                # If we do not have a job-user override & we're not explicitly running
+                # as the agent's user, then we *MUST* have a jobRunAsUser from the JobDetails.
+                # Reasons:
+                #  1) The service always allows service-managed Fleets to associate with
+                #     a Queue. The SMF Worker Agent is *always* run with a local user override.
+                #  2) The service only allows a customer-managed Fleet to associate with a
+                #     Queue if either:
+                #       a) The jobRunAsUser is explicitly set to WORKER_AGENT_USER; or
+                #       b) The jobRunAsUser is explicitly set to QUEUE_CONFIGURED_USER and
+                #          a user has been defined for the CMF's OS Platform (Linux/MacOS Fleets must
+                #          have a "posix" user; and Windows Fleets must have a "windows" user)
+                #  3) The service does not allow a Queue's jobRunAsUser to be updated if the constraint
+                #     imposed by (2) above would be violated for one or more of that Queue's current QFAs.
+                if host_is_posix:
+                    os_user = job_run_as_user.posix
+                else:
+                    os_user = job_run_as_user.windows
+                if os_user is None:
+                    # Terminal error. We need to fail the Session.
+                    # This should *never* happen; it occuring would mean that a service invariant has
+                    # been violated.
+                    message = (
+                        "FATAL: Queue's jobRunAsUser does not define a QUEUE_CONFIGURED_USER for this platform. "
+                        "Please report this to the service team."
+                    )
+                    raise ValueError(message)
+                else:
+                    logger.info(
+                        SessionLogEvent(
+                            subtype=SessionLogEventSubtype.USER,
+                            queue_id=queue_id,
+                            job_id=job_id,
+                            session_id=session_id,
+                            user=os_user.user,
+                            message="Running as Queue's jobRunAsUser.",
+                        )
+                    )
+        if os_user is None:
+            try:
+                user_to_log = getpass.getuser()
+            except Exception:
+                # This is best-effort. If we cannot determine the user we will not log
+                user_to_log = "UNKNOWN"
+            logger.warning(
```

Comment:
Not sure the user/operator has any insight why this is a warning log and not just an info. Should we swap it to info? Or add additional information?